db/python/tables/analysis.py

```diff
@@ -270,6 +287,79 @@ async def query(self, filter_: AnalysisFilter) -> List[AnalysisInternal]:
 
         return result
 
+    async def get_file_outputs_by_analysis_ids(
+        self, analysis_ids: list[int]
+    ) -> dict[int, dict[str, RecursiveDict]]:
+        """Fetches all output files for a list of analysis IDs"""
+
+        _query = """
+        SELECT ao.analysis_id, f.*, ao.json_structure, ao.output
+        FROM analysis_outputs ao
+        LEFT JOIN output_file f ON ao.file_id = f.id
+        WHERE ao.analysis_id IN :analysis_ids
+        """
+        rows = await self.connection.fetch_all(_query, {'analysis_ids': analysis_ids})
+
+        # Preparing to accumulate analysis files
+        analysis_files: dict[
+            int, dict[str, list[Tuple[OutputFileInternal, str]] | str]
+        ] = defaultdict(lambda: defaultdict(list))
+
+        for row in rows:
+            file_id = row['id']
+            if file_id:
+                # Building OutputFileInternal object with secondary files if available
+                file_internal = OutputFileInternal.from_db(**dict(row))
+                # If no json_structure, just set to the output.
+
+                if row['json_structure'] is None:
+                    analysis_files[row['analysis_id']]['output'] = row['path']
+                else:
+                    analysis_files[row['analysis_id']]['output'] = ''
+                analysis_files[row['analysis_id']]['outputs'].append((file_internal, row['json_structure']))  # type: ignore [union-attr]
+            else:
+                # If no file_id, just set to the output.
+                analysis_files[row['analysis_id']]['output'] = row['output']
+                analysis_files[row['analysis_id']]['outputs'] = row['output']
+
+        # Transforming analysis_files into the desired output format
+        analysis_output_files = {
+            a_id: {
+                'output': files['output'],
+                'outputs': OutputFileInternal.reconstruct_json(files['outputs']),
+            }
+            for a_id, files in analysis_files.items()
+        }
+
+        return analysis_output_files  # type: ignore [return-value]
+
+    async def get_secondary_files_for_file_output(
```

Comment:
This method isn't called anywhere, is it needed for something in the future? Or would it be ok to remove?